create_layout.py

```diff
@@ -91,47 +91,48 @@
 import in_toto.models.link
 import in_toto.models.layout
 
-def create_material_matchrules(links, index):
+def create_material_rules(links, index):
   """Create generic material rules (3 variants)
 
-  * No materials recorded -> disallow any artifact
-  * Materials recorded (first step) -> allow artifacts that existed beforehand
-  * Materials recorded (latter step) -> match from previous products
-
+  * MATCH available materials with products from previous step (links must be an
+  ordered list) and
+  * ALLOW available materials if it is the first step in the
+  list
   Returns a list of material rules
   NOTE: Read header docstring for ideas for more complexity.  """
 
-  material_matchrules = []
-
-  if not links[index].materials:
-    material_matchrules = [["DISALLOW", "*"]]
+  expected_materials = []
 
-  elif index == 0 and links[index].materials:
-    material_matchrules = [["ALLOW", "*"]]
+  if index == 0:
+    for material_name in links[index].materials.keys():
+      expected_materials.append(["ALLOW", material_name])
+    expected_materials.append(["DISALLOW", "*"])
 
   else:
-    material_matchrules = [
+    expected_materials = [
         ["MATCH", "*", "WITH", "PRODUCTS", "FROM", links[index - 1].name]]
 
-  return material_matchrules
+  return expected_materials
 
 
-def create_product_matchrules(links, index):
+def create_product_rules(links, index):
   """Create generic material rules (2 variants)
```

Comment:
s/material/product

Reply:
Well spotted, but not part of this PR. :P @SolidifiedRay, would you kindly fix this docstring when working on this code base?

Reply:
> Well spotted, but not part of this PR. :P @SolidifiedRay, would you kindly fix this docstring when working on this code base?

Yes, sure.